packages/@aws-cdk/aws-lambda/lib/scalable-function-attribute.ts

```diff
@@ -0,0 +1,88 @@
+import * as appscaling from '@aws-cdk/aws-applicationautoscaling';
+import { Construct, IConstruct, Token } from '@aws-cdk/core';
+
+/**
+ * Interface for scalable attributes
+ */
+export interface IScalableFunctionAttribute extends IConstruct {
+  /**
+   * Scale out or in to keep utilization at a given level. The utilization is tracked by the
+   * LambdaProvisionedConcurrencyUtilization metric, emitted by lambda. See:
+   * https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html#monitoring-metrics-concurrency
+   *
+   * Allowed values: 0.1 - 0.9.
+   */
+  scaleOnUtilization(options: UtilizationScalingOptions): void;
+  /**
+   * Scale out or in based on schedule.
+   */
+  scaleOnSchedule(id: string, actions: appscaling.ScalingSchedule): void;
+}
+
+/**
+ * Options for enabling Lambda utilization tracking
+ */
+export interface UtilizationScalingOptions extends appscaling.BaseTargetTrackingProps {
+  /**
+   * Utilization target for the attribute. For example, .5 indicates that 50 percent of allocated provisioned concurrency is in use.
+   *
+   * Allowed values: 0.1 - 0.9.
+   */
+  readonly utilizationTarget: number;
+}
+
+/**
+ * A scalable lambda alias attribute
+ */
+export class ScalableFunctionAttribute extends appscaling.BaseScalableAttribute {
```

Comment:
I believe this needs to extend `IScalableFunctionAttribute`, no?

Reply:
We want to extend the abstract class, but return the interface to the caller so only the `scaleOnxxx` methods will be exposed

Reply:
I agree with returning `IScalableFunctionAttribute` in the `addAutoScaling()` method. 

My question is if this should be -

```ts
export class ScalableFunctionAttribute extends appscaling.BaseScalableAttribute implements IScalableFunctionAttribute {
```

Keeps the type expectations in the correct place and any mismatches will be reported at the correct locations by the compiler.